docs/tuning-guide.md

```diff
@@ -64,20 +73,47 @@ the GPU and host memory as the transfer can be performed asynchronously from the
 pinned, so by default no pinned memory is allocated.
 
-It is recommended to use some amount of pinned memory when using the RAPIDS Accelerator if
-possible. Ideally the amount of pinned memory allocated would be sufficient to hold the input
+It is recommended to use some amount of pinned memory when using the RAPIDS Accelerator.
+Ideally the amount of pinned memory allocated would be sufficient to hold the input
 partitions for the number of concurrent tasks that Spark can schedule for the executor.
 
 Note that the specified amount of pinned memory is allocated _per executor_.  For example, if
 each node in the cluster has 4 GPUs and therefore 4 executors per node, a configuration setting
-of `spark.rapids.memory.pinnedPool.size=4G` will allocate a total of 16GB of memory on the system.
+of `spark.rapids.memory.pinnedPool.size=4G` will allocate a total of 16 GiB of memory on the
+system.
 
 When running on YARN, make sure to account for the extra memory consumed by setting
 `spark.executor.memoryOverhead` to a value at least as large as the amount of pinned memory
 allocated by each executor.  Note that pageable, off-heap host memory is used once the pinned
 memory pool is exhausted, and that would also need to be accounted for in the memory overhead
 setting.
 
+Pinned memory can also be used to speed up spilling from GPU memory if there is more data
+than can fit in GPU memory. [The spill storage](#spill-storage) is configured separately, but
+for best performance spill storage should be taken into account when allocating pinned memory.
+For example if I have 4 concurrent tasks per executor, each processing 1 GiB batches, I will
+want about 4 GiB of pinned memory to handle normal input/output, but if I am going to sort a
+large amount of data I might want to increase the pinned memory pool to 8 GiB and give 4 GiB to
+the spill storage, so it can use pinned memory too.
+
+## Spill Storage
+
+Configuration key: [`spark.rapids.memory.host.spillStorageSize`](configs.md#memory.host.spillStorageSize)
+
+Default value: `1g`
+
+This is the amount of host memory that is used to cache spilled data before it is flushed to disk.
+The GPU Accelerator employs different algorithms that allow it to process more data than can fit in
+the GPU's memory. We do not support this for all operations, and are constantly trying to add more.
+The way that this can work is by spilling parts of the data to host memory or to disk, and then
+reading them back in when needed. Spilling in general is more expensive than not spilling, but
+spilling to a slow disk can be especially expensive. If you see spilling to disk happening in
+your query, and you are not using GPU Direct storage for spilling, you may want to add more
+spill storage.
+
+You can configure this larger than the amount of pinned memory you have, and the Accelerator will
+end up using non-pinned memory when this happens.
```

Comment:
Technically the host memory store will use non-pinned memory when it detects there are less than the configured maximum bytes of host spill buffers being tracked in host memory, tries to allocate a new buffer from the pinned pool, and fails.  So even if the host spill store size is configured to be less than the pinned pool size, it's not guaranteed all host spills will be to pinned memory -- depends on the usage of the pinned pool by other entities.  Not sure we need to go into this much detail here, but making sure that distinction was understood.